### DISCUSSION TITLE
MovewapointStart, MoveWaypointEnd Enchancement

### DISCUSSION THREAD

**Discussion by deancochran** [maplibre/maplibre-gl-directions#201] [Ideas]
Hello,

I'm actively developing a routing application using OSM data, maptiler, OSRM, and svelte. I've enjoyed using this package for just over a month!

I noticed that it would be useful to update some reactive aspects of my application when a user starts and or finishes moving a waypoint. However, I did not find an event like "fetchroutestart, or fetchrouteend" for this case.

Are there current developments to add to the event types? I would be interested in helping develop these enhancements.

**Comment by smellyshovel** [comment]
@deancochran Hello. Yes, you're right, there are no `movewaypoint<start/end>` events at the moment. On the other hand, `movewaypoint` by itself may be considered as the "end" variant. But still, no way to detect the starting point, you're right with that.

Feel free to contribute!

**Comment by smellyshovel** [comment]
If you decide to implement the discussed feature, the only thing I'd like to mention before you begin, is that I'd appreciate if you considered a possibility to add an appropriate event not only to existing waypoints, but also to hoverpoints, i.e. when adding a new waypoint in between of existing ones.

Not stating that it's absolutely required, but may make a lot of sense. Maybe even a single even with different `type` (or whatever) attribute.

**Comment by smellyshovel** [comment]
Closing this one as inactive. Feel free to re-open in case there's something new to add :+1:.

**Comment by smellyshovel** [comment]
This is now available, starting from v0.9.0.
